roxie/ccd/ccdserver.cpp

```diff
@@ -4460,21 +4466,40 @@ class CRemoteResultAdaptor : implements IEngineRowStream, implements IFinalRoxie
                     loggingFlags |= LOGGING_CHECKINGHEAP;
                 if (ctx->queryWorkUnit())
                     loggingFlags |= LOGGING_WUID;
+                if (serializeTraceId)
+                    loggingFlags |= LOGGING_TRACEID;
                 if (debugContext)
-                {
                     loggingFlags |= LOGGING_DEBUGGERACTIVE;
-                    logInfo.append(loggingFlags).append(ctxTraceLevel);
+
+                logInfo.append(loggingFlags).append(ctxTraceLevel);
+
+                if (debugContext)
+                {
                     MemoryBuffer bpInfo;
                     debugContext->serialize(bpInfo);
                     bpInfo.append((__uint64)(memsize_t) &activity);
                     logInfo.append((unsigned short) bpInfo.length());
                     logInfo.append(bpInfo.length(), bpInfo.toByteArray());
                 }
-                else
-                    logInfo.append(loggingFlags).append(ctxTraceLevel);
+
+                if (loggingFlags & LOGGING_TRACEID)
+                {
+                    //Serialize in binary to minimize the data transferred
+                    char traceId[bytesTraceId];
+                    char spanId[bytesSpanId];
```

Comment:
I'm not sure if we'll ever care to communicate the issampled flag to the workers (probably if they ever propagate traceparent headers out to remote services). If we choose to, we can mimic the pre-estabilshed otel format: traceparent: 
00-<trace-id>-<span-id>-<trace-flags>
where trace-flags:  (2 hex digits) Bit flags to control tracing behavior.

The leading 2 hex digits and the delimiters are not needed.